src/documents/consumer.py

```diff
@@ -304,6 +307,64 @@ def run(self) -> str:
 
             self.log.debug(f"Detected mime type: {mime_type}")
 
+            # special zip handling
+            if mime_type == "application/zip":
+                filesToProcess = extract_zip_to_tempdir(
+                    self.working_copy,
+                    Path(tempdir.name),
+                    self.log,
+                )
+                zip_name = self.working_copy.name
+                persistent_dir = Path(settings.SCRATCH_DIR)
+                persistent_dir.mkdir(parents=True, exist_ok=True)
+                for file in filesToProcess:
+                    mime = mimetypes.guess_type(str(file))[0]
+                    parser_class = (
+                        get_parser_class_for_mime_type(mime) if mime else None
+                    )
+                    if parser_class is not None:
+                        # Rename file to zipname#originalname before moving
+                        new_name = f"{zip_name}#{file.name}"
+                        dest = persistent_dir / new_name
+                        shutil.move(str(file), str(dest))
+                        source = getattr(self, "source", None)
+                        pdf_title = Path(file.name).stem
+                        overrides = DocumentMetadataOverrides(
+                            filename=new_name,
+                            title=pdf_title,
+                        )
+                        input_doc = ConsumableDocument(
+                            source=source,
+                            original_file=dest,
+                        )
+                        # Import here to avoid circular import
+                        from documents.tasks import consume_file
+
+                        consume_file.delay(input_doc, overrides)
+                if tempdir:
+                    tempdir.cleanup()
```

Comment:
```suggestion
```

I think better to use a context manager here, e.g. `with tempfile.TemporaryDirectory() as tempdir:` which will automatically cleanup